kong/timing/context.lua

```diff
@@ -0,0 +1,207 @@
+local cjson         = require("cjson.safe")
+local utils         = require("kong.tools.utils")
+
+local ngx_get_phase = ngx.get_phase
+local ngx_re_gmatch = ngx.re.gmatch
+
+local math_floor    = math.floor
+local setmetatable  = setmetatable
+local table_insert  = table.insert
+local table_remove  = table.remove
+
+local assert        = assert
+
+local _M            = {}
+local _MT           = { __index = _M }
+
+
+function _M:enter_subcontext(name)
+  assert(name ~= nil, "name is required")
+  table_insert(self.sub_context_stack, self.current_subcontext)
+
+  if not self.current_subcontext.child then
+    self.current_subcontext.child = {}
+  end
+
+  if not self.current_subcontext.child[name] then
+    self.current_subcontext.child[name] = {}
+  end
+
+  self.current_subcontext = self.current_subcontext.child[name]
+  self.current_subcontext.____start____ = utils.get_updated_now_ms()
```

Comment:
Could we use the new time function `monotonic_msec()`? it may be better.
See: #11536.
```
local cur_msec = require("resty.core.time").monotonic_msec
  local function get_monotonic_ms()
    update_time()
    return cur_msec()
  end

```

Reply:
done